spec/requests/categories_controller_spec.rb

```diff
@@ -372,6 +372,82 @@
         response.parsed_body["category_list"]["categories"].map { |x| x["id"] },
       ).not_to include(uncategorized.id)
     end
+
+    describe "with serialized category custom fields" do
+      class CategoryPlugin < Plugin::Instance
+        attr_accessor :enabled
+        def enabled?
+          @enabled
+        end
+      end
+
+      fab!(:plugin) { CategoryPlugin.new }
+
+      before do
+        plugin.enabled = false
+        clear_custom_fields
+      end
+
+      def add_custom_fields
+        plugin.enabled = true
```

Comment:
You can define another context and add this into `before` hook.

Reply:
Sorry, not following on this one. Is this referring to just
```
plugin.enabled = true
```
or the whole `add_custom_fields` method?

Reply:
I found the way tests are defined a bit strange because you are always doing two things in each test: testing without and with custom fields. Ideally, these scenarios would represent different tests. I have added a commit that refactors the test.